veles/secrets/huggingfaceapikey/validator_test.go

```diff
@@ -0,0 +1,224 @@
+		req.URL.Host = testURL.Host
+	}
+	return http.DefaultTransport.RoundTrip(req)
+}
+
+// mockHuggingfaceServer creates a mock Huggingface API server for testing
+func mockHuggingfaceServer(t *testing.T, expectedKey string, statusCode int) *httptest.Server {
+	t.Helper()
+
+	return httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// Check if it's a GET request to the expected endpoint
+		if r.Method != http.MethodGet || r.URL.Path != "/api/whoami-v2" {
+			t.Errorf("unexpected request: %s %s, expected: GET /api/whoami-v2", r.Method, r.URL.Path)
+			http.Error(w, "not found", http.StatusNotFound)
+			return
+		}
+
+		// Check Authorization header
+		authHeader := r.Header.Get("Authorization")
+		if !strings.HasSuffix(authHeader, expectedKey) {
+			t.Errorf("expected Authorization header to end with key %s, got: %s", expectedKey, authHeader)
+		}
+
+		// Set response
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(statusCode)
+	}))
+}
+
+func TestValidator(t *testing.T) {
+	cases := []struct {
+		name        string
+		statusCode  int
+		want        veles.ValidationStatus
+		expectError bool
+	}{
+		{
+			name:       "valid_key",
+			statusCode: http.StatusOK,
+			want:       veles.ValidationValid,
+		},
+		{
+			name:       "invalid_key_unauthorized",
+			statusCode: http.StatusUnauthorized,
+			want:       veles.ValidationInvalid,
+		},
+		{
+			name:        "server_error",
+			statusCode:  http.StatusInternalServerError,
+			want:        veles.ValidationFailed,
+			expectError: true,
+		},
+		{
+			name:        "bad_gateway",
+			statusCode:  http.StatusBadGateway,
+			want:        veles.ValidationFailed,
+			expectError: true,
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			// Create a mock server
+			server := mockHuggingfaceServer(t, validatorTestKey, tc.statusCode)
+			defer server.Close()
+
+			// Create a client with custom transport
+			client := &http.Client{
+				Transport: &mockTransport{testServer: server},
+			}
+
+			// Create a validator with a mock client
+			validator := huggingfaceapikey.NewValidator(
+				huggingfaceapikey.WithClient(client),
+			)
+
+			// Create a test key
+			key := huggingfaceapikey.HuggingfaceAPIKey{Key: validatorTestKey}
+
+			// Test validation
+			got, err := validator.Validate(context.Background(), key)
+
+			// Check error expectation
+			if tc.expectError {
+				if err == nil {
+					t.Errorf("Validate() expected error, got nil")
+				}
+			} else {
+				if err != nil {
+					t.Errorf("Validate() unexpected error: %v", err)
+				}
+			}
+
+			// Check validation status
+			if got != tc.want {
+				t.Errorf("Validate() = %v, want %v", got, tc.want)
+			}
+		})
+	}
+}
+
+func TestValidator_ContextCancellation(t *testing.T) {
+	// Create a server that delays response
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		time.Sleep(100 * time.Millisecond)
```

Comment:
Prefer avoiding sleeps in tests - it's possible to cancel the context immediately, for example https://github.com/google/osv-scalibr/blob/cda7ee373dfdb4926cef6b2a9e16ab8f4fb33da7/veles/secrets/gcpsak/validator_test.go#L189